ts/src/connections/tunnelConnectionBase.ts

```diff
@@ -168,9 +168,11 @@ export class TunnelConnectionBase implements TunnelConnection {
     /**
      * Throws CancellationError if the tunnel connection is disposed.
      */
-    protected throwIfDisposed() {
+    protected throwIfDisposed(message: string, stack: string|undefined = undefined) {
         if (this.isDisposed) {
-            throw new ObjectDisposedError('The tunnel connection is disposed.');
+            const error = new ObjectDisposedError(`The tunnel connection is disposed. ${message}`);
+            error.stack = stack;
```

Comment:
Only set it if `stack` is defined? Otherwise, can it wipe out existing `stack` on `error` object?